### ISSUE TITLE
Agent Forwarding

### ISSUE THREAD

**Issue by isontheline** [isontheline/pro.webssh.net#754]
Hi, does WebSSH support Agent Forwarding? It looks like it doesn’t but I thought I would check in case there’s something I’m doing wrong...

If not, please consider this a feature request!

Regards,
Matt

**Comment by isontheline** [comment]
Hi Matt 👋

Unfortunately no : 
https://github.com/isontheline/pro.webssh.net/issues/513

I will open an issue in order to track your request.

Have a nice day ☀️

**Comment by isontheline** [comment]
Thanks for the quick reply!

I just worked out I can achieve what I need using a wildcard tunnel as a bastion. 

So Agent Forwarding would be a nice to have but not necessary.

Thanks for your work on WebSSH, it’s a great app!

Regards,
Matt

**Comment by isontheline** [comment]
> Hey! Great works! Webssh is by far the best ssh client on iOS I have tried. But I want to share some improvement ideas with you.
> 
> 2. Please support ssh key agent forward. This is definitely an premium feature. I would pay any price for this! Keep your key local and forwarding key request on remote machine is a good practice for security and probably the only way to work in production environment.
>  Thanks you!

**Comment by haskelious** [comment]
Agent Forwarding would definitely help in managing servers more securely. From my perspective this is the only feature I miss to fully move over from Termius.

**Comment by isontheline** [comment]
Bonjour Stephane @haskelious 👋

I will try to implement it.

Right now a possible workaround is to use the Dynamic Port Forwarding inside WebSSH : it will allow you to connect through your bastion host like you could with Agent Forwarding.

Have you already tried it?

Bonne journée à vous ☀️

**Comment by unnilennium** [comment]
from my understanding Dynamic Port Forwarding will only help for first hop.  not it you need to chain multiple servers to access the last resource.

**Comment by isontheline** [comment]
You're right @unnilennium , so you could use the new feature available inside WebSSH : "Connect through" :

1. Launch a SSH connection to host1
2. Launch a SSH connection to host2 by using the contextual menu "Connect through"
3. etc.

**Comment by unnilennium** [comment]
well this will not solve dns this way and force you to input an ip, but as the ip is not routable from where i am i get:

        <error>Failed to open direct-tcpip channel</error>

if i put a hostname fails ti resolve it.  

only on the first hop. 

thank you for the suggestions, but

nothing can beat the simplicity and efficiency of forwardagent ;)

**Comment by isontheline** [comment]
@unnilennium Are you on WebSSH 29.1?

**Comment by unnilennium** [comment]
> @unnilennium Are you on WebSSH 29.1?

yes

**Comment by Nyxtorm** [comment]
+1, this would be a really useful feature.

It would be great to introduce independent key management for User-Agent handling. In particular, being able to:

- Choose which key is used for the connection, with the option to add it to the agent or not
- Select additional keys to load into the agent separately

Ideally, this could be configurable with a priority order:

- Global level
- Group level
- Connection level

This would provide fine-grained control over authentication behavior and better support more complex setups.

**Comment by isontheline** [comment]
Thanks for your strong feedback @Nyxtorm 🙏
Hope I can work on it
